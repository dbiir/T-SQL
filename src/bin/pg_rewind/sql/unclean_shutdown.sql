#!/bin/bash

# This file has the .sql extension, but it is actually launched as a shell
# script. This contortion is necessary because pg_regress normally uses
# psql to run the input scripts, and requires them to have the .sql
# extension, but we use a custom launcher script that runs the scripts using
# a shell instead.

TESTNAME=unclean_shutdown

. sql/config_test.sh

MASTER_PG_CTL_STOP_MODE="immediate"

# The old master was stopped with immediate mode which will cause unclean
# shutdown and leave "in production" in the control file. At the beginning of
# pg_rewind, a single-user mode postgres session should have been run to ensure
# clean shutdown on the target instance.

# Run the test
. sql/run_test.sh
