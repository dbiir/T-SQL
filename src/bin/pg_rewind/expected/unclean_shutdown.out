Master initialized.
Master running.
Standby initialized and running.
Standby promoted.
Running pg_rewind...
Old master restarted after rewind.
SELECT state FROM pg_stat_replication;
   state   
-----------
 streaming
(1 row)

Master promoted.
